salt/metalk8s/orchestrate/deploy_node.sls

```diff
@@ -70,7 +72,29 @@ Cordon the node:
   metalk8s_cordon.node_cordoned:
     - name: {{ node_name }}
 
-{%- if not pillar.orchestrate.get('skip_draining', False) %}
+{%- if not skip_draining or cleanup_loop_devices %}
+  {%- set run_drain = not skip_draining %}
+
+  {%- if skip_draining %}
+    {# Check if we can avoid the drain (if no loop device needs cleaning up) #}
+    {%- set volumes = salt.metalk8s_kubernetes.list_objects(
+      kind="Volume", apiVersion="storage.metalk8s.scality.com/v1alpha1",
+    ) %}
+    {%- set volumes_to_clean = volumes
+                             | selectattr('spec.nodeName', 'equalto', node_name)
+                             | selectattr('spec.sparseLoopDevice', 'defined')
+                             | list
+    %}
+    {%- if volumes_to_clean %}
+      {%- do salt.log.warning(
+        'Forcing drain to clean up loop devices for the following Volumes: '
+        ~ volumes_to_clean | map(attribute='metadata.name') | join(', ')
+      ) %}
+      {%- set run_drain = True %}
```

Comment:
Do we really want to force ?
Could be better/safer to just fail hard at this point, no ?

Reply:
Not sure: if we fail hard, it means we can never upgrade a single-node cluster with sparse volumes (unless we remove the volumes before upgrade, which is cumbersome). Since such clusters only exist for dev purposes, I prefer to force the drain (and create a service interruption) rather than prevent the upgrade.

I guess another option could be to require some `metalk8s:upgrade:force_drain` pillar value or something along those lines, but I'm not sure it's worth the effort. WDYT @scality/metalk8s ?

Reply:
Hmm yes ok you right make sense.